Dumbness shouldn't drive...

MystLeissa opened the pull request:
Prevents people with the Dumb Trait from piloting mecha

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

Split off from RC-1 this isolation contains the idea that it's a slight oversight that people with the dumb trait can drive a highly complicated machine. We'll fix that
 (Note: I have no IDEA where to get the TRAIT_DUMB from naturally but I used it in my lockable schoolgirl branch to interact with this as well as a few other things that the flag came up on)

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

Piloting a Mech Suit shouldn't be so simple a dumb person can do it.

## Changelog
:cl:
tweak: Dumb People Can't Drive Mecha anymore.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

mystleissa2-boop committed:
Dumbness shouldn't drive...

Prevents people with the Dumb Trait from piloting mecha

zxaber commented:
Would be cooler if having the dumb trait automatically applied the uncontrollable movement that you normally get from mech malfunctions.

MystLeissa commented:
> Would be cooler if having the dumb trait automatically applied the uncontrollable movement that you normally get from mech malfunctions.

Was thinking about that myself - but I'm not sure that's within my skill level...it isn't exactly as easy as setting an my_pos + <frand(3),frand(3),frand(3)>

HomocidlePancakes commented:
Should do the same to savages

Fikou reviewed (changes requested):
this has bad spelling and random whitespaces for no reason please fix

nemvar commented:
@HomocidlePancakes Savages trait isn't real.

MystLeissa commented:
> this has bad spelling and random whitespaces for no reason please fix

where are words misspelled? As for white space - that's a matter of preference - I used tab - and any other whitespace is from me hitting enter so it's more of a linebreak.

MrDoomBringer commented on code/game/mecha/mecha.dm:
```diff
@@ -949,8 +954,7 @@
 			return mmi_moved_inside(mmi_as_oc, user)
 		else
 			to_chat(user, "<span class='warning'>Occupant detected!</span>")
-	else
```
Why did this else get deleted?

MystLeissa replied:
it was not needed, I'd have to reference the entire proc to know for sure - but I'd say the reasoning was the "Protip: Early Returns are preferred" reference in the wiki.

MystLeissa replied:
Actually your right - I fixed it  in the latest commit; I'm pretty sleepless..

mystleissa2-boop committed:
Added Missing span close tag

For pities sake - I need sleep and I have more important things to do than chase down rouge white space

kingofkosmos commented:
Could this be applied through relaymove to driving anything at all?
Also text could have a little fluff like "You have no idea how to operate \the [src]!"

mystleissa2-boop committed:
Added Back Missing Else

Else was removed from proc due to user error and sleeplessness. This adds it back.

Akrilla commented on code/game/mecha/mecha.dm:
```diff
@@ -892,9 +892,11 @@
 		return
-
+	if(HAS_TRAIT(user,TRAIT_DUMB))
+		to_chat(user, "<span class='warning'>This Mech will not let dumb people drive it</span>")
```
Mech isn't capitalized, also needs some sort of punctuation at the end.

Akrilla commented on code/game/mecha/mecha.dm:
```diff
@@ -904,11 +906,12 @@
 		else if(user.has_buckled_mobs())
 			to_chat(user, "<span class='warning'>You can't enter the exosuit with other creatures attached to you!</span>")
+		else if(HAS_TRAIT(user,TRAIT_DUMB))
+			to_chat(user,"<span class='warning'>You can't drive dummy</span>")
```
same here

Akrilla commented on code/game/mecha/mecha.dm:
```diff
@@ -904,11 +906,12 @@
-		to_chat(user, "<span class='warning'>You stop entering the exosuit!</span>")
-	return
+			return
+		to_chat(user, "<span class='warning'>You stop entering the exosuit!</span>");
```
You've removed a return here, so this line will _always_ be said, which I don't imagine is intended.

MrPerson replied:
The return is if the move into the mecha was successful. By this point, the only case left is failure of some type. It's inverted logic, but lolmechs

ATH1909 commented:
Don't we already have a "dumb" brain trauma? Why not just check for that (or the clumsy trait/mutation)?

Detective-Google commented:
> Don't we already have a "dumb" brain trauma? Why not just check for that (or the clumsy trait/mutation)?

if you check for clumsy clowns couldn't pilot H.O.N.K.
